tests/functional/fixtures/authentication.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+
+__all__ = (
+    "user",
+    "with_logged_in_user",
+    "with_logged_in_staff_member",
+    "with_logged_in_admin",
+)
+
+
+@pytest.fixture
+def user(factories):
+    return factories.User()
+
+
+@pytest.fixture
+def with_logged_in_user(login_user):
+    login_user()
+
+
+@pytest.fixture
+def with_logged_in_staff_member(login_user, user):
+    user.staff = True
+
+    login_user()
+
+
+@pytest.fixture
+def with_logged_in_admin(login_user, user):
+    user.admin = True
+
+    login_user()
+
+
+@pytest.fixture
+def login_user(db_session, app, user):
+    def login_user():
+        # This is the hash for `pass` used below
+        user.password = "$2b$12$21I1LjTlGJmLXzTDrQA8gusckjHEMepTmLY5WN3Kx8hSaqEEKj9V6"
+        db_session.commit()
```

Comment:
This commit means any changes made to the user prior to this call take effect, like the `user.admin = True` above as well as setting the password.

The original had the password set on creation, but I think I prefer this version as it keeps the setting of the hash and using the value in the same place. Previously it was a bit obscure as to why this was being set.